packages/shared/src/permissions/permissionsClass.ts

```diff
@@ -643,6 +643,96 @@ export class Permissions {
     );
   }
 
+  public canReadSingleProjectResource = (
+    project: string | undefined
+  ): boolean => {
+    return this.hasPermission("readData", project || "");
+  };
+
+  public canReadMultiProjectResource = (
+    projects: string[] | undefined
+  ): boolean => {
+    // If the resource is available to all projects (an empty array or no projects property), then everyone should have read access
+    if (!projects || !projects.length || this.superAdmin) {
+      return true;
+    }
+
+    const hasGlobalReadAccess =
```

Comment:
Can we simplify all of the following logic to this one-liner?

```ts
return projects.some(p=>this.hasPermission("readData", project))
```

Reply:
It can - I didn't think that would handle the logic of the last `readData` test suite test - `User with global readonly role should not be able to see metrics if the user has noaccess permission for every one of the metrics projects` but after testing, the code above does handle it correctly.